snakefmt/parser/parser.py

```diff
@@ -324,3 +325,12 @@ def get_next_queriable(self, snakefile: TokenIterator) -> Status:
             if not pythonable and token.type != tokenize.COMMENT:
                 pythonable = True
             buffer += token.string
+            if (
+                token is not None
+                and sys.version_info >= (3, 12)
+                and token.type == tokenize.FSTRING_MIDDLE
+            ):
+                if token.string.endswith("}"):
+                    buffer += "}"
+                elif token.string.endswith("{"):
+                    buffer += "{"
```

Comment:
v. nice fix; I noticed this was almost identical to the equivalent fix in parameter parsing in `syntax.py`, so I proposed a refactoring, on tip of master on [my fork](https://github.com/bricoletc/snakefmt)

(commit: bb56cddb49f55296e11a66eb7600009c96fd130a)